qa/L0_batcher/batcher_test.py

```diff
@@ -38,16 +38,43 @@
 import infer_util as iu
 import test_util as tu
 from tensorrtserver.api import *
+import tensorrtserver.shared_memory as shm
 import tensorrtserver.api.server_status_pb2 as server_status
+from ctypes import *
 
 if os.environ['BATCHER_TYPE'] == "VARIABLE":
     _trials = ("savedmodel", "graphdef", "netdef", "custom", "libtorch", "onnx")
 else:
     _trials = ("savedmodel", "graphdef", "plan", "netdef", "custom", "libtorch", "onnx")
 
+if "TEST_SHARED_MEMORY" in os.environ:
+    TEST_SHARED_MEMORY=int(os.environ["TEST_SHARED_MEMORY"])
+else:
+    TEST_SHARED_MEMORY=0
+
 _max_queue_delay_ms = 10000
 _check_exception = None
 
+def _create_advance(shm_regions = None):
+    if TEST_SHARED_MEMORY:
+        precreated_shm_regions = []
+        shared_memory_ctx = SharedMemoryControlContext("localhost:8000", ProtocolType.HTTP, verbose=True)
+        if shm_regions is None:
+            shm_regions = ['output0','output1']
+        for shm_region in shm_regions:
+            shm_tmp_handle = shm.create_shared_memory_region(shm_region +'_data', '/'+ shm_region, 512)
+            precreated_shm_regions.append(shm_tmp_handle)
+            shared_memory_ctx.register(shm_tmp_handle)
+        return precreated_shm_regions
+    return shm_regions
```

Comment:
Yes, the intention is that. In the case TEST_SHARED_MEMORY is not set the shared memory regions are not created/registered. When that is the case a string on the names is passed to prevent the indexing for the precreated_shm_regions to not fail.

Reply:
Fixed